File: Python/pythonPractice/snakeWaterGun.py
import random
import time

def compare(you , comp):
	if you == comp:
		return 0
	elif you == 0 and comp == 1:
		return 1
	elif you == 1 and comp == 0:
		return -1
	elif you == 0 and comp == 2:
		return -1
	elif you == 2 and comp == 0:
		return 1
	elif you == 1 and comp == 2:
		return 1
	elif you == 2 and comp == 1:
		return -1

name = input("Please enter your name.\n")
print("Hello",name,"\n\t\tWelcome to 'Snake Water and Gun' game")

start = input("\nPress c to continue\n")
if start.capitalize() == 'C':
	try:
		Your_score = 0
		CPU_score = 0
		list= ["Snake", "Water", "Gun"]
		i = 1
		while(i !=4):
			print("Round no.",i,"\nYours turn....")
			player = int(input("Press 0 for Snake, 1 for Water and 2 for Gun\n"))
			print("\nYou have choosen",list[player],"\n")
	
			print("CPU turns....")
			time.sleep(1)
			computer = random.randint(0, 2)
			print("CPU has choosen",list[computer],"\n")
			result = compare(player, computer)
			if result == 0:
				print("Its a draw !\n")
			elif result == 1:
				print("You got it.\n")
				Your_score += 1
			elif result == -1:
				print("CPU got it.\n")
				CPU_score += 1
			i =  i + 1
		if Your_score == CPU_score:
			print("Match draw !!\n")
		elif Your_score > CPU_score:
			print("Congrats",name,"You win.\n")
		else:
			print("Sorry",name,"You lose\n")
		print("Your score:",Your_score)
		print("CPU score:",CPU_score)
		print("\nThanks for playing..")
	except Exception:
		print("Something went wrong.Please try again...")
	
else:
	print("Something went wrong. please try again...")

File: C/Snake water gun.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

int Random(int n)
{
    srand(time(NULL));
    return rand()%n;
}

void Fullname(char c)
{
    if(c == 's')
    {
        printf("Choosen 'Snake'\n\n");
    }
    else if(c == 'w')
    {
        printf("Choosen 'Water'\n\n");
    }
    else if(c == 'g')
    {
        printf("Choosen 'Gun'\n\n");
    }
    else
    {
        printf("Error!..you choose invalid character\n");
    }

}

int compResult(char you, char cpu)
{
    if(you == cpu)
    {
        return 0;
    }
    else if(you == 's' && cpu == 'w')
    {
        return 1;
    }
    else if(you == 'w' && cpu == 's')
    {
        return -1;
    }
    else if(you == 's' && cpu == 'g')
    {
        return -1;
    }
    else if(you == 'g' && cpu == 's')
    {
        return 1;
    }
    else if(you == 'w' && cpu == 'g')
    {
        return 1;
    }
    else if(you == 'g' && cpu == 'w')
    {
        return -1;
    }
}

int main()
{
    char you, cpu,n;
    int player_score = 0, cpu_score = 0;
    for(int i = 0; i < 3; i++)
    {
        printf("Round no.%d\n",i+1);
        printf("Yours turn...\n");
        printf("Enter 's' for snake, 'w' for water and 'g' for gun\n");
        scanf("%c",&you);
        getchar();
        Fullname(you);

        printf("CPU turns...\n");
        n = Random(100);
        if(n<33)
        {
            cpu = 's';
        }
        else if(n>33 && n<66)
        {
            cpu = 'w';
        }
        else
        {
            cpu = 'g';
        }
        Fullname(cpu);

        int result = compResult(you, cpu);
        if(result == 0)
        {
            printf("Its a draw!\n");
        }
        else if(result == 1)
        {
            player_score+=1;
            printf("You got it!\n");
        }
        else
        {
            cpu_score+=1;
            printf("CPU got it!\n");
        }
        printf("Your score:%d\nCPU score:%d\n\n",player_score, cpu_score);
    }

    if(player_score == cpu_score)
    {
        printf("Match draw.\n");
    }
    else if(player_score > cpu_score)
    {
        printf("You win.\n");
    }
    else
    {
        printf("You lose.\n");
    }


    return 0;
}
